src/kfactory/routing/optical.py

```diff
@@ -125,8 +127,8 @@ def route(
     c: KCell,
     p1: Port,
     p2: Port,
-    straight_factory: Callable[[int, int], KCell],
-    bend90_cell: KCell,
+    straight_factory: Callable[..., KCell] = straight,
+    bend90_cell: KCell | Callable[..., KCell] = bend_euler,
```

Comment:
what if you pass enclosures?

Reply:
The routing function cannot apply enclosures. So, even if you were to add it as a kwargs, in case of no default it fails and in case of a default, it applies the default.

So, the reasoning behind this, probably makes sense if you think of a bend as independent from the straoght waveguide:

In the past we had it that the enclosure for bends needs to have larger excludes than the straights, because the mode could be bigger at some parts of the bend than on straights.

As for how to apply, use partials or lambdas or similar.

I'd be happy to write some example cases for kgeneric if we don't have it yet.